.github/workflows/haskell-ci.yml

```diff
@@ -14,8 +14,12 @@
 #
 name: Haskell-CI
 on:
-  - push
-  - pull_request
+  push:
+    branches:
+      - master
```

Comment:
should we just disable pushing to master?

Reply:
We already did; the goal of this is to CI check the result of merging to master (which is not necessarily the CI check of any given PR since PRs may be based on older master).